### PR TITLE
Access logic reverted to old version + added tests

### LINKED ISSUE #5132: Drawing during multi access on R&D messes up following accesses
no further text, because it's already fixed

### PR DESCRIPTION
**Pull request by lostgeek**
Closes #5132

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
That’s a good test.

File: test/clj/game/core/runs_test.clj [hunk 1]
```diff
@@ -440,3 +440,154 @@
...
+          (card-ability state :corp sc 1)
+          (click-prompt state :corp "Efficiency Committee")
+          ; R&D was shuffled
+          (click-prompt state :runner "No") ; not letting sub fire
+          (click-prompt state :runner "No action") ; end accessing C
+          (is (= (str "You accessed " (-> (get-corp) :deck first :title) ".")
+                 (-> (get-runner) :prompt first :msg))
+              "Accessing top card of R&D")
+          (click-prompt state :runner "No action")
+          (is false "no")
```
Comment (NoahTheDuke) [inline test/clj/game/core/runs_test.clj:592]: 👀

File: test/clj/game/core/runs_test.clj [hunk 2]
```diff
@@ -440,3 +440,154 @@
...
+          (click-prompt state :runner "No action")
+          (is (= "You accessed Brainstorm." (-> (get-runner) :prompt first :msg)) "Accessed B")
+          (click-prompt state :runner "No action")
+          (is (= "You are encountering Chrysalis. Allow its subroutine to fire?" (-> (get-runner) :prompt first :msg)) "Accessing C")
+          (card-ability state :corp sc 1)
+          (click-prompt state :corp "Efficiency Committee")
+          ; R&D was shuffled
+          (click-prompt state :runner "No") ; not letting sub fire
+          (click-prompt state :runner "No action") ; end accessing C
+          (is (= (str "You accessed " (-> (get-corp) :deck first :title) ".")
```
Comment (NoahTheDuke) [inline test/clj/game/core/runs_test.clj:588]: This fails if after shuffling you encounter Chrysalis again.

File: test/clj/game/core/runs_test.clj (hunk 1 above)
Comment (lostgeek) [reply test/clj/game/core/runs_test.clj:592]: Whoops. I was using that line to make the test fail so I can see the output even if it would succeed :joy:

**Pull request merged**